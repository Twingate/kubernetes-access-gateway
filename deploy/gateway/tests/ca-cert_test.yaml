suite: Certificate Authority Certificate
templates:
  - ca-cert.yaml
set:
  tls:
    autoGenerated:
      enabled: true
      engine: cert-manager
tests:
  - it: should not render when `autoGenerated` is disabled
    set:
      tls:
        autoGenerated:
          enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should not render when `engine` is helm
    set:
      tls:
        autoGenerated:
          engine: helm
    asserts:
      - hasDocuments:
          count: 0
  - it: should not render when `existingSecret` is provided
    set:
      tls:
        existingSecret: "my-tls-secret"
    asserts:
      - hasDocuments:
          count: 0
  - it: should render using self-signed CA
    asserts:
      - matchSnapshot: {}
  - it: should use `existingIssuer` when provided
    set:
      tls:
        autoGenerated:
          enabled: true
          engine: cert-manager
          certManager:
            existingIssuer: existing-issuer
            existingIssuerKind: ClusterIssuer
    asserts:
      - equal:
          path: spec.issuerRef.name
          value: existing-issuer
        documentSelector:
          path: kind
          value: Certificate
      - equal:
          path: spec.issuerRef.kind
          value: ClusterIssuer
        documentSelector:
          path: kind
          value: Certificate
      - hasDocuments:
          # 1 Certificate + 1 Issuer
          count: 2
