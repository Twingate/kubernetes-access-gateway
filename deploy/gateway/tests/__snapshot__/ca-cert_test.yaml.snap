should render using self-signed CA:
  1: |
    apiVersion: cert-manager.io/v1
    kind: Issuer
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway-clusterissuer
    spec:
      selfSigned: {}
  2: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway-ca-crt
    spec:
      commonName: RELEASE-NAME-kubernetes-access-gateway-ca
      isCA: true
      issuerRef:
        kind: Issuer
        name: RELEASE-NAME-kubernetes-access-gateway-clusterissuer
      secretName: RELEASE-NAME-kubernetes-access-gateway-ca-crt
  3: |
    apiVersion: cert-manager.io/v1
    kind: Issuer
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway-ca-issuer
    spec:
      ca:
        secretName: RELEASE-NAME-kubernetes-access-gateway-ca-crt
